From 00a9ec84b6f50fd602a18ae4372c408ca1c24361 Mon Sep 17 00:00:00 2001
From: Sam Thursfield <sam.thursfield@codethink.co.uk>
Date: Fri, 16 Jun 2017 16:30:43 +0000
Subject: Add patch to allow compilation against musl toolchains

Beware that this probably breaks compilation on any other platform,
it's not a real solution to the issue.

https://sourceware.org/bugzilla/show_bug.cgi?id=21604
---
 sunrpc/rpc/types.h | 22 ++++++----------------
 1 file changed, 6 insertions(+), 16 deletions(-)

diff --git a/sunrpc/rpc/types.h b/sunrpc/rpc/types.h
index beded52..4a71c17 100644
--- a/sunrpc/rpc/types.h
+++ b/sunrpc/rpc/types.h
@@ -69,24 +69,14 @@ typedef unsigned long rpcport_t;
 #include <sys/types.h>
 #endif
 
-#if defined __APPLE_CC__ || defined __FreeBSD__
-# define __u_char_defined
-# define __daddr_t_defined
-#endif
-
-#ifndef __u_char_defined
-typedef __u_char u_char;
-typedef __u_short u_short;
-typedef __u_int u_int;
-typedef __u_long u_long;
-typedef __quad_t quad_t;
-typedef __u_quad_t u_quad_t;
-typedef __fsid_t fsid_t;
+/* IMPORTANT NOTE: This has been modified to build against the musl C
+ * library and it probably now ONLY builds with the musl C library.
+ *
+ * See: https://sourceware.org/bugzilla/show_bug.cgi?id=21604
+ */
 # define __u_char_defined
-#endif
 #ifndef __daddr_t_defined
-typedef __daddr_t daddr_t;
-typedef __caddr_t caddr_t;
+typedef int daddr_t;
 # define __daddr_t_defined
 #endif
 
-- 
cgit v0.12-1-gdc88

